staging/src/k8s.io/cli-runtime/pkg/resource/query_param_verifier_v3.go

```diff
@@ -106,6 +106,9 @@ func hasGVKExtensionV3(extensions spec.Extensions, gvk schema.GroupVersionKind)
 // the PATCH end-point. Returns true if the query param is supported by the
 // spec for the passed GVK; false otherwise.
 func supportsQueryParamV3(doc *spec3.OpenAPI, gvk schema.GroupVersionKind, queryParam VerifiableQueryParam) bool {
+	if doc == nil || doc.Paths == nil {
+		return false
```

Comment:
does this fallback to checking v2 openapi for fieldValidation support or to client-side validation in this case?

Reply:
I think this needs to return a NotFound error to trigger the fallback to secondary in fallbackQueryParamVerifier#HasSupport

Reply:
no, that returns an "unsupported" error, which is positively asserting we scanned the openapi and determined the query parameter is not supported. That does not trigger this fallback:

https://github.com/kubernetes/kubernetes/blob/51cfa6134a2e357c6b30b6c4ad7dba1ce887bc9d/staging/src/k8s.io/cli-runtime/pkg/resource/fallback_query_param_verifier.go#L50-L54

A nil v3 doc or a v3 doc with no paths should fall back to checking openapi v2

Reply:
I thought that only openapi/v3 or openapi/v2 can exist in the cluster and in that case it is openapi/v3(although it is nearly empty `Response Body: {"openapi":"3.0.0","info":{"title":"allocation.agones.dev"},"components":{}}` but that does not mean that http status: 404) and no need to fallback to v2. Because it can not be existed. It seems that I'm wrong.

Reply:
Yeah, if we have an empty openapi v3, it's quite unlikely that we'll have a full openapi v2 (because the aggregator converts v2 into v3 when v3 is missing). But an aggregated apiserver could public a valid, good openapi v2 and a bad openapi v3, in which case we should probably still falback to v2.

> I thought that only openapi/v3 or openapi/v2 can exist in the cluster and in that case it is openapi/v3(although it is nearly empty Response Body: {"openapi":"3.0.0","info":{"title":"allocation.agones.dev"},"components":{}} but that does not mean that http status: 404) and no need to fallback to v2. Because it can not be existed. It seems that I'm wrong.

Yeah, we can totally have both, and as I mentioned above, it's rather the opposite: you can't have v2 without v3 ;-)